### PR TITLE
[Hub Generated] Review request for Microsoft.Cache to add version preview/2023-05-01-preview

### PR DESCRIPTION
**Pull request by koderjoker**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/koderjoker-redis-Microsoft.Cache-2023-05-01-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by TimLovellSmith** [comment]
>             }

202 status should not normally return a payload, schema should just be omitted.

---
Refers to: specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json:1817 in 4a3cacc. [](commit_id = 4a3cacc7988118145f00547f7fdf468631e6e9c9, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             }

same, not needed

---
Refers to: specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json:2028 in 4a3cacc. [](commit_id = 4a3cacc7988118145f00547f7fdf468631e6e9c9, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             }

is it also possible to disable non-aad-auth?

---
Refers to: specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json:2246 in 4a3cacc. [](commit_id = 4a3cacc7988118145f00547f7fdf468631e6e9c9, deletion_comment = False)

**Comment by v-xuto** [comment]
@koderjoker Please fix CI checks errors (`LintDiff`, `Acocado`), you can refer to this [comment](https://github.com/Azure/azure-rest-api-specs/pull/23745#issuecomment-1525076133) or https://aka.ms/ci-fix on how to fix it.

**Comment by koderjoker** [comment]
@v-xuto please ignore these errors, the avacado errors are irrelevant to this api version and the Lintdiff errors are due to an issue with the pipeline as quoted here https://github.com/Azure/azure-rest-api-specs/pull/23406#issuecomment-1496788714

**Comment by koderjoker** [comment]
@TimLovellSmith we got CI failures for not returning a payload so we had to add it. authnotrequired was in the previous swagger version as well so we should allow customers to do so without aad

**Comment by v-xuto** [comment]
@qiaozha Please help review JS breaking change.
@raych1  Please help review Go breaking change.

**Comment by v-xuto** [comment]
@live1206 Please help review `azure-sdk-for-net-track2` [error](https://github.com/Azure/azure-rest-api-specs/pull/23745#issuecomment-1525059740).

**Comment by live1206** [comment]
> @live1206 Please help review `azure-sdk-for-net-track2` [error](https://github.com/Azure/azure-rest-api-specs/pull/23745#issuecomment-1525059740).

It failed due to breaking changes, but not introduced in this PR.

**Comment by TimLovellSmith** [comment]
@live1206 The breaking changes, for addition of regex pattern validation, are approved. So what else needs to be done?

**Comment by TimLovellSmith** [comment]
> e got CI failures for not returning a payload so we had to add it. authnotrequired was in the previous swagger version as well so we should allow customers to do so without aad

Authnotrequired doesn't mean aad-auth only, it turns off authentication completely.

**Comment by TimLovellSmith** [comment]
@koderjoker 
>we got CI failures for not returning a payload 
Where?

**Comment by koderjoker** [comment]
> @koderjoker
> 
> > we got CI failures for not returning a payload
> > Where?

Sorry, misunderstood the previous statement as removing it from the specification only. Removed the body from delete 202 response now. Thanks for pointing this out @TimLovellSmith

File: specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json [hunk 1]
```diff
@@ -0,0 +1,3286 @@
...
+    "CacheNameParameter": {
+      "name": "cacheName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the Redis cache.",
+      "x-ms-parameter-location": "method",
+      "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$"
+    },
+    "AccessPolicyNameParameter": {
```
Comment (qiaozha) [inline specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json:3265]: Please add x-ms-parameter-location  method to this parameter definition like line 3262

File: specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json [hunk 2]
```diff
@@ -0,0 +1,3286 @@
...
+      "name": "accessPolicyName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the access policy that is being added to the Redis cache.",
+      "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$",
+      "minLength": 3,
+      "maxLength": 63
+    },
+    "AccessPolicyAssignmentNameParameter": {
```
Comment (qiaozha) [inline specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json:3275]: the same for this parameter

File: specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json (hunk 1 above)
Comment (koderjoker) [reply specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json:3265]: Added method

File: specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json (hunk 2 above)
Comment (koderjoker) [reply specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01-preview/redis.json:3276]: Added method

**Comment by koderjoker** [comment]
@live1206 @raych1 could you please help us get the approved labels?

**Comment by v-xuto** [comment]
@raych1 Please help review Go breaking change.

**Comment by v-xuto** [comment]
@koderjoker  Please resolve the conflict.
image

**Comment by koderjoker** [comment]
@v-xuto resolved the conflict

**Review by v-xuto — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**